src/utils/pgwire/src/pg_protocol.rs

```diff
@@ -173,29 +172,16 @@ where
                     .collect();
 
                 // 2. Create the row description.
-                let rows: Vec<PgFieldDescriptor> = query
-                    .split(&[' ', ',', ';'])
-                    .skip(1)
-                    .into_iter()
-                    .map(|x| {
-                        if let Some(str) = x.strip_prefix('$') {
-                            if let Ok(i) = str.parse() {
-                                i
-                            } else {
-                                -1
-                            }
-                        } else {
-                            -1
-                        }
-                    })
-                    .take_while(|x: &i32| x.is_positive())
-                    .map(|x| {
-                        // NOTE Make sure the type_description include all generic parametre
-                        // description we needed.
-                        assert!((x.sub(1) as usize) < types.len());
-                        PgFieldDescriptor::new(String::new(), types[x.sub(1) as usize].to_owned())
-                    })
-                    .collect();
```

Comment:
Why these part of logic looks so complicated while the impl in Session is so short...

Reply:
Because the original implement needs to infer the row description from the types descriptions. Now, We just simply get the row description from the infer_return_type. The complicated parts are left to the implementation of infer_return_type.